stdlib/public/core/ContiguousArrayBuffer.swift

```diff
@@ -490,7 +492,13 @@ internal func += <Element, C : Collection>(
       from: lhs.firstElementAddress, count: oldCount)
     lhs.count = 0
     swap(&lhs, &newLHS)
-    (lhs.firstElementAddress + oldCount).initialize(from: rhs)
+    buf = UnsafeMutableBufferPointer(start: lhs.firstElementAddress + oldCount, count: numericCast(rhs.count))
+  }
+
+  if let remainder = buf.initialize(from: rhs) {
+    // there were elements that didn't fit in the existing buffer,
+    // append them in slow sequence-only mode
+    //FIXME: handle this possibility
```

Comment:
Ah, you're right, this function is taking a collection so underflow should trap.